src/app.ts

```diff
@@ -30,23 +24,43 @@ app.use('/viewer', viewerRouter);
 
 const server = createServer(app);
 
-server.listen(process.env['VIV_PORT'], async () => {
-    await Promise.all(asyncInits.map(async (i) => await i()));
-    console.log(`App is listening on port ${process.env['VIV_PORT']}!`);
-});
-
 let shutdownTimer: NodeJS.Timeout | null = null;
 export const { clientsAt, messageClientsAt } = setupSockets(
     server,
     () => {
-        const timeout = parseInt(process.env['VIV_TIMEOUT'] ?? '10000');
-        if (timeout > 0)
+        if (config.timeout > 0)
             shutdownTimer = setInterval(() => {
-                console.log(`No clients for ${timeout}ms, shutting down.`);
+                console.log(`No clients for ${config.timeout}ms, shutting down.`);
                 process.exit(0);
-            }, timeout);
+            }, config.timeout);
     },
     () => {
         if (shutdownTimer) clearInterval(shutdownTimer);
     },
 );
+
+const address = `http://localhost:${config.port}`;
+const openArgs = () => {
+    for (const path of process.argv.slice(2)) {
+        if (path.startsWith('-')) continue;
+        if (!existsSync(path)) {
+            console.log(`File not found: ${path}`);
+            continue;
+        }
+        const absolute = presolve(path);
+        const url = `${address}${pathToURL(absolute)}`;
+        open(url);
+    }
+};
+
+get(`${address}/health`, () => {
+    // server is already running
+    openArgs();
+    process.exit(0);
```

Comment:
`process.exit(0)` needs to wait that `openArgs()` is finished (specifically `open(url)` if I've understood correctly)

Reply:
Oh, I didn't realize this was async. My bad. Will fix it later.

Reply:
Can you check if it works now? I hopefully fixed it in https://github.com/jannis-baum/vivify/pull/90/commits/b72355a7ca374e7fd362d67953b5d68dc6eb1e72 :)

Reply:
Yes, fixed! Thank you